lib/membrane_rtc_engine/endpoints/webrtc/rtp_munger/cache.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Membrane.RTC.Engine.Endpoint.WebRTC.RTPMunger.Cache do
+  @moduledoc false
+
+  # Cache for RTPMunger
+  # Stores information about mapping of sequence numbers from one space to another
+  # It is a moving window with a size defined by difference of first and last sequence number present in the cache
+
+  # !!! WARNING !!!
+  # This module doesn't handle out-of-order entries, there isn't even a chech as it's not meant to be used outside RTPMunger.
+  # RTPMunger is responsible for ensuring that entries are added in appropriate order
+
+  use Bunch.Access
+
+  @max_seq_num 2 ** 16
+  @history_size div(@max_seq_num, 8)
+
+  @enforcekeys [:cache]
+  defstruct @enforcekeys
+
+  @type t() :: %__MODULE__{
+          cache: Qex.t({non_neg_integer(), non_neg_integer(), non_neg_integer()})
+        }
+
+  @spec new() :: t()
+  def new(), do: %__MODULE__{cache: Qex.new()}
+
+  @spec push(t(), non_neg_integer(), non_neg_integer()) :: t()
+  def push(%__MODULE__{} = state, from, to) do
+    state
+    |> Map.update!(:cache, &Qex.push(&1, {from, to}))
+    |> remove_outdated_entries()
+  end
+
+  @spec get(t(), non_neg_integer()) :: {:ok, non_neg_integer()} | {:error, :not_found}
+  def get(%__MODULE__{cache: cache}, from) do
+    cache
+    |> Enum.find(fn
+      {^from, _to} -> true
+      _otherwise -> false
+    end)
+    |> case do
+      nil ->
+        {:error, :not_found}
+
+      {^from, to} ->
+        {:ok, to}
+    end
+  end
+
+  defp remove_outdated_entries(state) do
+    window_size = get_window_size(state)
+
+    if window_size >= @history_size do
+      {_entry, cache} = Qex.pop(state.cache)
+      remove_outdated_entries(%{state | cache: cache})
+    else
+      state
+    end
+  end
```

Comment:
There is a problem with naming here, consider such scenario

100, 101, 110, 111, 112, ..., 65 535, 0, 1, 2, ..., 200, 201, 210, 211, 212, ...

We will notice two gaps, the first one from 102 to 109 and then the second one from 202 to 209. When adding the second range to the cache we will invoke `remove_outdated_entries` function but it won't remove 102 to 109 gap despite the fact it is clearly outdated. I think that the whole solution will work but it could be a little bit clearer. Names such as `history_size`, `remove_outdated` `window_size` don't mean things I would expect them to mean.

Reply:
No, this will not happen, because we remove outdated entries whenever a new entry is added. Therefore, 102 to 109 will be long gone by the time you even get to 65535. The only way that this could occur is if you a massive gap in sequence numbers, which, as we discussed many times, is not detectable for us (mayby apart from looking at a wall clock, but we don't do that kind of thing)